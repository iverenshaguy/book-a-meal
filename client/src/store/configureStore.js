import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rootReducer, { history } from '../reducers/rootReducer';

const enhancers = [];
const middlewares = [thunk, routerMiddleware(history)];


if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);

  const { devToolsExtension } = window;

  if (typeof devToolsExtension === 'function') {
    enhancers.push(devToolsExtension());
  }
}

const composedEnhancers = compose(applyMiddleware(...middlewares), ...enhancers);

export default createStore(rootReducer, composedEnhancers);
