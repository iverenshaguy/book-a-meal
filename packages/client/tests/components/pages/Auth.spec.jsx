import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import AuthComponent from '../../../src/components/pages/Auth';
import AuthContainer from '../../../src/containers/pages/Auth';
import { initialState } from '../../setup/mockData';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);

const unAuthStore = mockStore({
  ...initialState,
  auth: { ...initialState.auth, isAuthenticated: false }
});

const props = {
  submitting: false,
  isAuthenticated: false,
  submitError: null
};

const signinLocation = {
  pathname: '/signin',
  state: { from: { pathname: '/' } },
};

const catererSignupLocation = {
  pathname: '/signup',
  state: { from: { pathname: '/' } },
  search: '?role=caterer'
};

const customerSignupLocation = {
  pathname: '/signup',
  state: { from: { pathname: '/' } },
  search: '?role=customer'
};
const { now } = Date;

describe('Auth', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render Signin Form component correctly', () => {
    const wrapper = shallow(<AuthComponent {...props} location={signinLocation} type="signin" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render customer\'s signup form component correctly', () => {
    const wrapper = shallow(<AuthComponent {...props} location={customerSignupLocation} type="signup" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render signin component that is connected to the redux store correctly', () => {
    const comp = (<Provider store={unAuthStore}><AuthContainer {...props} location={signinLocation} type="signin" /></Provider>);
    const mountedWrapper = mount(comp, rrcMock.get());

    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });

  it('should set the form component state to signin when the type prop equals signin', () => {
    const comp = (<AuthComponent {...props} location={customerSignupLocation} type="signin" />);
    const wrapper = shallow(comp);

    expect(wrapper.state().type).toEqual('signin');
  });

  it('should set the form component state to customerSignup when the type prop equals customerSignup', () => {
    const comp = (<AuthComponent {...props} location={customerSignupLocation} type="signup" />);
    const wrapper = shallow(comp);

    expect(wrapper.state().type).toEqual('customerSignup');
  });

  it('should set the form component state to catererSignup when the type prop equals catererSignup', () => {
    const comp = (<AuthComponent {...props} location={catererSignupLocation} type="signup" />);
    const wrapper = shallow(comp);

    expect(wrapper.state().type).toEqual('catererSignup');
  });

  it('should render caterer\'s signup form correctly when role query is caterer', () => {
    const wrapper = shallow(<AuthComponent {...props} location={catererSignupLocation} type="signup" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should redirect use to the homepage when the user is already authenticated', () => {
    const comp = (<AuthContainer store={store} {...props} location={catererSignupLocation} type="catererSignup" />);
    const wrapper = mount(comp, rrcMock.get());

    expect(wrapper.find('Redirect')).toBeTruthy();
    expect(wrapper.find('Redirect').props().to).toEqual('/');
  });
});
