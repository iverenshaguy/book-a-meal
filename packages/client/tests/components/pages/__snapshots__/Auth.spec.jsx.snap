// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should render Signin Form component correctly 1`] = `
<Fragment>
  <Header
    dateType={null}
    logout={null}
    showTime={true}
    type="unauth"
    updateCurrentDate={null}
    user={null}
  />
  <div
    className="main-wrapper auth-wrapper"
  >
    <div
      className="auth"
    >
      <div
        className="title"
      >
        <h2>
          Welcome Back
        </h2>
        <hr
          className="w-75"
        />
      </div>
      <Connect(Form)
        isAuthenticated={false}
        location={
          Object {
            "pathname": "/signin",
            "state": Object {
              "from": Object {
                "pathname": "/",
              },
            },
          }
        }
        meta={
          Object {
            "btnText": "SIGN IN",
            "extra": <React.Fragment>
              <p
                className="form-extra-info forgot-password"
              >
                <Link
                  replace={false}
                  to="/forgot-password"
                >
                  Forgot Password?
                </Link>
              </p>
              <p
                className="form-extra-info text-center"
              >
                <React.Fragment>
                  Don't have an account?
                   
                  <Link
                    replace={false}
                    to="/signup?role=customer"
                  >
                    Signup Here
                  </Link>
                </React.Fragment>
              </p>
              <p
                className="text-center"
              >
                Or
              </p>
              <p
                className="form-extra-info text-center"
              >
                <Link
                  replace={false}
                  to="/signup?role=caterer"
                >
                  Signup as a Caterer
                </Link>
              </p>
            </React.Fragment>,
          }
        }
        submitError={null}
        submitting={false}
        type="signin"
      />
    </div>
  </div>
  <Footer />
</Fragment>
`;

exports[`Auth should render caterer's signup form correctly when role query is caterer 1`] = `
<Fragment>
  <Header
    dateType={null}
    logout={null}
    showTime={true}
    type="unauth"
    updateCurrentDate={null}
    user={null}
  />
  <div
    className="main-wrapper auth-wrapper"
  >
    <div
      className="auth"
    >
      <div
        className="title"
      >
        <h2>
          Start Serving Customers
        </h2>
        <hr
          className="w-75"
        />
      </div>
      <Connect(Form)
        isAuthenticated={false}
        location={
          Object {
            "pathname": "/signup",
            "search": "?role=caterer",
            "state": Object {
              "from": Object {
                "pathname": "/",
              },
            },
          }
        }
        meta={
          Object {
            "btnText": "SIGN UP",
            "extra": <React.Fragment>
              <p
                className="form-extra-info text-center"
              >
                <React.Fragment>
                  Already have an account?
                   
                  <Link
                    replace={false}
                    to="/signin"
                  >
                    Sign In Here
                  </Link>
                </React.Fragment>
              </p>
              <p
                className="text-center"
              >
                Or
              </p>
              <p
                className="form-extra-info text-center"
              >
                <Link
                  replace={false}
                  to="/signup?role=customer"
                >
                  Signup as a Customer
                </Link>
              </p>
            </React.Fragment>,
          }
        }
        submitError={null}
        submitting={false}
        type="catererSignup"
      />
    </div>
  </div>
  <Footer />
</Fragment>
`;

exports[`Auth should render customer's signup form component correctly 1`] = `
<Fragment>
  <Header
    dateType={null}
    logout={null}
    showTime={true}
    type="unauth"
    updateCurrentDate={null}
    user={null}
  />
  <div
    className="main-wrapper auth-wrapper"
  >
    <div
      className="auth"
    >
      <div
        className="title"
      >
        <h2>
          Start Filling Your Belly
        </h2>
        <hr
          className="w-75"
        />
      </div>
      <Connect(Form)
        isAuthenticated={false}
        location={
          Object {
            "pathname": "/signup",
            "search": "?role=customer",
            "state": Object {
              "from": Object {
                "pathname": "/",
              },
            },
          }
        }
        meta={
          Object {
            "btnText": "SIGN UP",
            "extra": <React.Fragment>
              <p
                className="form-extra-info text-center"
              >
                <React.Fragment>
                  Already have an account?
                   
                  <Link
                    replace={false}
                    to="/signin"
                  >
                    Sign In Here
                  </Link>
                </React.Fragment>
              </p>
              <p
                className="text-center"
              >
                Or
              </p>
              <p
                className="form-extra-info text-center"
              >
                <Link
                  replace={false}
                  to="/signup?role=caterer"
                >
                  Signup as a Caterer
                </Link>
              </p>
            </React.Fragment>,
          }
        }
        submitError={null}
        submitting={false}
        type="customerSignup"
      />
    </div>
  </div>
  <Footer />
</Fragment>
`;

exports[`Auth should render signin component that is connected to the redux store correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Auth)
    isAuthenticated={false}
    location={
      Object {
        "pathname": "/signin",
        "state": Object {
          "from": Object {
            "pathname": "/",
          },
        },
      }
    }
    submitError={null}
    submitting={false}
    type="signin"
  >
    <Auth
      dispatch={[Function]}
      isAuthenticated={false}
      location={
        Object {
          "pathname": "/signin",
          "state": Object {
            "from": Object {
              "pathname": "/",
            },
          },
        }
      }
      submitError={null}
      submitting={false}
      type="signin"
    >
      <Header
        dateType={null}
        logout={null}
        showTime={true}
        type="unauth"
        updateCurrentDate={null}
        user={null}
      >
        <div
          className="header unauth"
        >
          <div
            className="navbar unauth-navbar"
          >
            <div
              className="page-title"
            >
              <h3>
                <Link
                  replace={false}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    BOOK-A-MEAL
                  </a>
                </Link>
              </h3>
            </div>
          </div>
        </div>
      </Header>
      <div
        className="main-wrapper auth-wrapper"
      >
        <div
          className="auth"
        >
          <div
            className="title"
          >
            <h2>
              Welcome Back
            </h2>
            <hr
              className="w-75"
            />
          </div>
          <Connect(Form)
            dispatch={[Function]}
            isAuthenticated={false}
            location={
              Object {
                "pathname": "/signin",
                "state": Object {
                  "from": Object {
                    "pathname": "/",
                  },
                },
              }
            }
            meta={
              Object {
                "btnText": "SIGN IN",
                "extra": <React.Fragment>
                  <p
                    className="form-extra-info forgot-password"
                  >
                    <Link
                      replace={false}
                      to="/forgot-password"
                    >
                      Forgot Password?
                    </Link>
                  </p>
                  <p
                    className="form-extra-info text-center"
                  >
                    <React.Fragment>
                      Don't have an account?
                       
                      <Link
                        replace={false}
                        to="/signup?role=customer"
                      >
                        Signup Here
                      </Link>
                    </React.Fragment>
                  </p>
                  <p
                    className="text-center"
                  >
                    Or
                  </p>
                  <p
                    className="form-extra-info text-center"
                  >
                    <Link
                      replace={false}
                      to="/signup?role=caterer"
                    >
                      Signup as a Caterer
                    </Link>
                  </p>
                </React.Fragment>,
              }
            }
            submitError={null}
            submitting={false}
            type="signin"
          >
            <Form
              dispatch={[Function]}
              email={null}
              isAuthenticated={false}
              location={
                Object {
                  "pathname": "/signin",
                  "state": Object {
                    "from": Object {
                      "pathname": "/",
                    },
                  },
                }
              }
              meal={null}
              meta={
                Object {
                  "btnText": "SIGN IN",
                  "extra": <React.Fragment>
                    <p
                      className="form-extra-info forgot-password"
                    >
                      <Link
                        replace={false}
                        to="/forgot-password"
                      >
                        Forgot Password?
                      </Link>
                    </p>
                    <p
                      className="form-extra-info text-center"
                    >
                      <React.Fragment>
                        Don't have an account?
                         
                        <Link
                          replace={false}
                          to="/signup?role=customer"
                        >
                          Signup Here
                        </Link>
                      </React.Fragment>
                    </p>
                    <p
                      className="text-center"
                    >
                      Or
                    </p>
                    <p
                      className="form-extra-info text-center"
                    >
                      <Link
                        replace={false}
                        to="/signup?role=caterer"
                      >
                        Signup as a Caterer
                      </Link>
                    </p>
                  </React.Fragment>,
                }
              }
              submitError={null}
              submitting={false}
              token={null}
              type="signin"
              uploading={null}
            >
              <div>
                <form
                  onSubmit={[Function]}
                >
                  <SigninForm
                    handlers={
                      Object {
                        "handleBlur": [Function],
                        "handleChange": [Function],
                        "handleFocus": [Function],
                        "handleSubmit": [Function],
                      }
                    }
                    state={
                      Object {
                        "error": Object {
                          "email": null,
                          "password": null,
                        },
                        "formValid": false,
                        "pristine": true,
                        "touched": Object {
                          "email": false,
                          "password": false,
                        },
                        "type": "signin",
                        "typingTimeout": 0,
                        "values": Object {
                          "email": "",
                          "password": "",
                        },
                      }
                    }
                    type="signin"
                  >
                    <RenderInput
                      handleBlur={[Function]}
                      handleChange={[Function]}
                      handleFocus={[Function]}
                      id="email"
                      label="Email Address"
                      meta={
                        Object {
                          "asyncValidating": undefined,
                          "error": null,
                          "touched": false,
                        }
                      }
                      name="email"
                      placeholder=""
                      required={true}
                      type="email"
                      value=""
                    >
                      <div
                        className="form-input"
                      >
                        <label
                          htmlFor="email"
                          onClick={[Function]}
                        >
                          Email Address
                          <span
                            className="danger"
                          >
                            *
                          </span>
                        </label>
                        <input
                          className=""
                          id="email"
                          name="email"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          placeholder=""
                          type="email"
                          value=""
                        />
                         
                      </div>
                    </RenderInput>
                    <RenderInput
                      handleBlur={[Function]}
                      handleChange={[Function]}
                      handleFocus={[Function]}
                      id="password"
                      label="Password"
                      meta={
                        Object {
                          "error": null,
                          "touched": false,
                        }
                      }
                      name="password"
                      placeholder=""
                      required={true}
                      type="password"
                      value=""
                    >
                      <div
                        className="form-input"
                      >
                        <label
                          htmlFor="password"
                          onClick={[Function]}
                        >
                          Password
                          <span
                            className="danger"
                          >
                            *
                          </span>
                        </label>
                        <input
                          className=""
                          id="password"
                          name="password"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          placeholder=""
                          type="password"
                          value=""
                        />
                         
                      </div>
                    </RenderInput>
                  </SigninForm>
                  <button
                    className="btn btn-pri btn-block"
                    disabled={true}
                    type="submit"
                  >
                    SIGN IN
                  </button>
                </form>
                <p
                  className="form-extra-info forgot-password"
                >
                  <Link
                    replace={false}
                    to="/forgot-password"
                  >
                    <a
                      href="/forgot-password"
                      onClick={[Function]}
                    >
                      Forgot Password?
                    </a>
                  </Link>
                </p>
                <p
                  className="form-extra-info text-center"
                >
                  Don't have an account?
                   
                  <Link
                    replace={false}
                    to="/signup?role=customer"
                  >
                    <a
                      href="/signup?role=customer"
                      onClick={[Function]}
                    >
                      Signup Here
                    </a>
                  </Link>
                </p>
                <p
                  className="text-center"
                >
                  Or
                </p>
                <p
                  className="form-extra-info text-center"
                >
                  <Link
                    replace={false}
                    to="/signup?role=caterer"
                  >
                    <a
                      href="/signup?role=caterer"
                      onClick={[Function]}
                    >
                      Signup as a Caterer
                    </a>
                  </Link>
                </p>
              </div>
            </Form>
          </Connect(Form)>
        </div>
      </div>
      <Footer>
        <div
          className="footer"
        >
          <p>
            © 2018. Iveren Shaguy. All images were gotten from the internet and are not mine.
          </p>
        </div>
      </Footer>
    </Auth>
  </Connect(Auth)>
</Provider>
`;
